src/schema/v2/infiniteDiscovery/discoverArtworks.ts

```diff
@@ -39,14 +39,21 @@ export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
       defaultValue: 2,
     },
   }),
-  resolve: async (_root, args, { artworksDiscoveryLoader }) => {
+  resolve: async (_root, args, { artworksDiscoveryLoader, meLoader }) => {
+    if (!meLoader || !artworksDiscoveryLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    const me = await meLoader()
```

Comment:
I'm surprised that we need to request `me` to get the requesting user's ID. If we can fetch `me` doesn't that mean that we already know who they are?

Reply:
agree, there are two parts:

- I probably can take user id without making a call to `me` - let me check if I can do it
- Gravity requires user_id argument - I think we should rewrite it a bit - make `user_id` param optional and use `current_user` by default (and overwrite it with a user from `user_id`, if needed). 

cc @egdbear

Reply:
Yeah, the issue with meLoader is that discoverArtworks is on the root type, rather than being part of the User type where the 'authorization' is already handled. Adding discoverArtworks to the User type should resolve this without needing meLoader. 
Then the query would look like this:

```
query {
    user { 
        discoverArtworks()...
    }
}
```